docs/Reference-for-template.json.md

```diff
@@ -52,31 +52,275 @@ You can find a sample of a complex `template.json` file here [Sample](https://gi
 |exclude|Exclude configuration specific to this sources.modifiers instance, contingent on the corresponding sources.modifiers.condition. See sources.exclude for more info|
 |copyOnly|CopyOnly configuration specific to this sources.modifiers instance, contingent on the corresponding sources.modifiers.condition. See sources.copyonly for more info|
 
-#### Symbols Introduction
+### Symbols 
 The symbols section defines variables and their values, the values may be the defined in terms of other symbols. When a defined symbol name is encountered anywhere in the template definition, it is replaced by the value defined in this configuration. The symbols configuration is a collection of key-value pairs. The keys are the symbol names, and the value contains key-value-pair configuration information on how to assign the symbol a value.
-
-|Name|Description|Applies to|
-|--|--|--|
-|type|Defines the high level configuration of symbol. The valid types are: `parameter`, `computed`, and `generated`. The type indicates how the configuration is processed to generate the symbol value:||
-||`Parameter`: A symbol for which the config provides literal and/or default values||
-||`Computed`: A symbol for which the config provides a Boolean predicate whose evaluation result is the computed symbol result||
-||`Generated`: A symbol whose value gets computed by a built-in symbol value generator. See [Details](https://github.com/dotnet/templating/wiki/Reference-for-available-macros)||
-|dataType |Indicates limitations on the valid values a symbol may be assigned. At this point, the only valid datatype is "choice", which also requires providing symbols.choices configuration for the symbol|`type=parameter`|
-|defaultValue|The value assigned to the symbol if no value for it is provided by the user or host|`type=parameter`|
-|binding|||
-|replaces|Text value that will be replaced||
-|fileRename|The fileRename element defines the filenames which will be replaced|`type=parameter`|
-|description|Human readable text describing the meaning of the symbol. This has no effect on template generation||
-|isRequired|Indicates if the parameter is required or not.|Only used when `type` is set to `parameter`|
-|choices|List of available choices.|Only used when `type` is `choice`|
-|value |Value for replacement|`type=computed`|
+There are the following symbol types:
+- Parameter
+- Derived
+- Computed
+- Generated
+- Bind
 
 There are 3 places from which a symbol can acquire its value:
+- Template configuration (this).
+- Host provided values, which override template configuration values. For example, the host may provide the operating system kind as "Linux", overriding a template default value of "Windows".
```

Comment:
Is there any documentation on host provider values? What are they?

Reply:
Seems like pretty basic info to be left out of the documentation.

Reply:
I agree. @KathleenDollard this is one example that I had in mind.